<launch>
    <arg name="rand" default="10.0" />
    <arg name="viz" default="false" />
    <arg name="output" default="screen" />
    <arg name="cgraph" default="true" />
    <arg name="bag" default="true" />

    <node pkg="swarm_localization" name="swarm_localization" type="swarm_localization_node" output="$(arg output)">
        <rosparam>
            force_freq: 10
            max_accept_cost: 100
            max_keyframe_num: 10000
            min_keyframe_num: 1
            thread_num: 20
            min_kf_movement : 0.5
            init_xy_movement : 1.0
            init_z_movement : 0.5
            pub_swarm_odom: true
            loop_outlier_threshold_pos: 1.0
            loop_outlier_threshold_yaw: 0.5
            enable_detection: true
            enable_detection_depth: false
            enable_loop: true
            enable_distance: false
            vo_drift_meter: 0.03
            vo_drift_meter_z: 0.05
            vo_drift_angle: 0.02
            distance_measurement_error: 0.2
            loop_cov_xy: 0.5
            loop_cov_z: 0.5
            loop_cov_yaw: 0.5
            detection_sphere_cov: 0.2
            detection_inv_dep_cov: 0.8
            publish_full_path: true
            is_pc_replay: true
            kf_use_all_nodes: false
        </rosparam>
        <param name="swarm_nodes_config" value="$(find swarm_localization)/config/swarm_nodes5.yaml" type="string" />
        <param name="initial_random_noise" value="$(arg rand)" type="double" />
        <param name="cgraph_path" value="$(env HOME)/graph.dot" type="string" />
        <param name="enable_cgraph_generation" value="$(arg cgraph)" type="bool" />
        <remap from="/swarm_drones/swarm_drone_fused" to="/swarm_drones/swarm_drone_fused_pc" />
    </node>
    <group if="$(arg viz)">
        <node name="rvizvisualisation" pkg="rviz" type="rviz" output="log" args="-d $(find swarm_localization)/config/5drone.rviz" />
    </group>
    <node pkg="rosbag" type="record" name="rosbag_record_local" 
        args="record -O /home/xuhao/swarm_ws/src/swarm_localization/swarm_localization/DataAnalysis/data/swarm_local_pc.bag -a" 
        if="$(arg bag)" />
</launch>
